app/features-json/setup_json_controller.rb

```diff
@@ -5,8 +5,40 @@ module FastlaneCI
   class SetupJSONController < APIController
     HOME = "/data/setup"
 
-    get "#{HOME}/configured", authenticate: false do
-      return json(Services.onboarding_service.correct_setup?)
+    get "#{HOME}/configured_sections", authenticate: false do
+      return json({
+        encryption_key: !FastlaneCI.dot_keys.encryption_key.nil?,
+        # Setting the OAuth keys without a provider type is not robust, we should
+        # find a way to differentiate client ID/Secret by provider type (Ex. GitHub)
+        oauth: !FastlaneCI.dot_keys.oauth_client_id.nil? && !FastlaneCI.dot_keys.oauth_client_secret.nil?,
+        config_repo: !FastlaneCI.dot_keys.repo_url.nil?
+      })
+    end
+
+    post "#{HOME}/encryption_key", authenticate: false do
+      key = params["encryption_key"]
+
+      if key.nil?
+        json_error!(
+          error_message: "Must provide encryption key",
+          error_key: "EncryptionKey.Missing",
+          error_code: 400
+        )
+      end
+
+      unless FastlaneCI.dot_keys.encryption_key.nil?
+        json_error!(
+          error_message: "The encryption key has already been set",
+          error_key: "EncryptionKey.AlreadyExists",
+          error_code: 403
+        )
+      end
+
+      Services.dot_keys_variable_service.write_keys_file!(
```

Comment:
Actually i've been using that as a reference. That PR was all the sections in one massive request. Since we're breaking it up, we'll need separate end points. You actually reminded me I forgot to add a character limit. I'll address that in another PR.

https://github.com/fastlane/ci/pull/1066/files#diff-a9e0bbe5fcea12f320816a0848465771R72